### PR TITLE
Implement new StartOperation(activity) overload

### PR DESCRIPTION
**Pull request by lmolkova**
Fix Issue #644 .
Introduces `telemetryClient.StartOperation(Activity)` overload.

It's expected to be used to track custom operations such as Queue messages processing
```C#
var message = queue.Receive();
var activity = message.ExtractActivity();

using (telemetryClient.StartOperation(activity))
{
  //process
}
```

Documentation update task - https://github.com/Microsoft/ApplicationInsights-dotnet/issues/676

### CODE REVIEW

File: src/Microsoft.ApplicationInsights/TelemetryClientExtensions.cs [hunk 1]
```diff
@@ -176,5 +176,63 @@ public static IOperationHolder<T> StartOperation<T>(this TelemetryClient telemet
 
             operation.Dispose();
         }
+
+        /// <summary>
+        /// Creates an operation object with a respective telemetry item.
```
Comment (SergeyKanzhelev) [inline src/Microsoft.ApplicationInsights/TelemetryClientExtensions.cs]: Please expand - explain whether Activity expected to be already started, which properties should already be initialized, etc.

File: Test/Microsoft.ApplicationInsights.Test/Standalone/AppInsightsStandaloneTests.cs [hunk 2]
```diff
@@ -87,7 +86,7 @@ static void Main(string[] args)
                         var tc = new TelemetryClient();
                         using (var requestOperation = tc.StartOperation<RequestTelemetry>(""request"", args[0]))
                         {
-                            using (var dependencyOperation = tc.StartOperation<DependencyTelemetry>(""dependency""))
+                            using (var dependencyOperation = tc.StartOperation<DependencyTelemetry>(""dependency"", args[0]))
```
Comment (SergeyKanzhelev) [inline Test/Microsoft.ApplicationInsights.Test/Standalone/AppInsightsStandaloneTests.cs]: can you please test both overloads.

Actually - one of the most important things for me is that `TrackTrace` will not crash due to `OperationCorrelationTelemetryInitializer` initialization failure without DS present

Comment (lmolkova) [reply Test/Microsoft.ApplicationInsights.Test/Standalone/AppInsightsStandaloneTests.cs]: The thing is, that StartOperation<DependencyTelemetry>("dependency") does not compile any longer without DS being referenced.

For .NET project, as AppInsights SDK is installed/updated, DiagnosticSource automatically by nuget.

The only problem is non .NET (non-nuget) scripts that could still use AI. As I mention here: https://github.com/Microsoft/ApplicationInsights-dotnet/issues/644#issuecomment-349754193
it's extremely unlikely that such scripts use StartOperation at all, as it's a generic method and could be called from proweshell only with reflection.

As for the StartOperation overloads: all of them except Activity-one are internally calling the same StartOperation(T telemetry), so they all use Activity underneath (when available).

I added OperationCorrelationTelemetryInitializer into the configuration and call to TrackTrace.

**Comment by lmolkova** [comment]
@cijothomas 

Could you please help me understand build failure? It failed in Codecov.ps1 with errors like 
```2017-12-07T19:31:03.0296713Z An System.IO.DirectoryNotFoundException occured: Could not find a part of the path 'E:\A\_work\7\s\src\Core\Managed\shared\OperationTelemetryExtensions.cs'. 
2017-12-07T19:31:03.0296713Z An System.IO.DirectoryNotFoundException occured: Could not find a part of the path 'E:\A\_work\7\s\src\Core\Managed\shared\TelemetryClient.cs'. 
```
I'm almost sure it has nothing to do with my changes.

@Dmitry-Matveev @MS-TimothyMothra @cijothomas 
please review

**Comment by cijothomas** [comment]
@lmolkova yes the unit tests actually passed on regualr run, but failed when ran with CodCov turned on. I have kicked off a re-run to see if this was transient issue.
(Surely this does not prevent the PR from merging)

**Comment by lmolkova** [comment]
@Dmitry-Matveev @MS-TimothyMothra 
please review

**Comment by lmolkova** [comment]
@cijothomas could you please merge it?

**Pull request merged**